crates/sui-graphql-rpc/src/types/dynamic_field.rs

```diff
@@ -172,66 +169,126 @@ impl DynamicField {
         };
 
         let field_id = derive_dynamic_field_id(parent, &type_, &name.bcs.0)
-            .map_err(|e| Error::Internal(format!("Failed to derive dynamic field id: {e}")))?
-            .to_vec();
+            .map_err(|e| Error::Internal(format!("Failed to derive dynamic field id: {e}")))?;
+
+        let key = match parent_version {
+            Some(version) => ObjectLookupKey::LatestAtParentVersion {
+                version,
+                checkpoint_viewed_at,
+            },
+            None => match checkpoint_viewed_at {
+                Some(seq) => ObjectLookupKey::LatestAt(seq),
+                None => ObjectLookupKey::Latest,
+            },
+        };
```

Comment:
nit (optional): I'm a fan of the match on tuple pattern for this kind of thing, but this is personal preference, this looks fine to be honest:

```suggestion
        use ObjectLookupKey as K;
        let key = match (parent_version, checkpoint_viewed_at) {
            (None, None) => K::Latest,
            (None, Some(checkpoint_viewed_at)) => K::LatestAt(checkpoint_viewed_at),
            (Some(version), checkpoint_viewed_at) => K::LatestAtParentVersion {
                version,
                checkpoint_viewed_at,
            },
        }
```

Reply:
lets do it